test/blackbox/api/dds-pim/PubSubReader.hpp

```diff
@@ -349,26 +349,27 @@ class PubSubReader
                 participant_qos_,
                 &participant_listener_,
                 eprosima::fastdds::dds::StatusMask::none());
-            ASSERT_NE(participant_, nullptr);
-            ASSERT_TRUE(participant_->is_enabled());
         }
 
-        participant_guid_ = participant_->guid();
+        if (participant_ != nullptr)
+        {
+            participant_guid_ = participant_->guid();
 
-        type_.reset(new type_support());
+            type_.reset(new type_support());
 
-        // Register type
-        ASSERT_EQ(participant_->register_type(type_), ReturnCode_t::RETCODE_OK);
+            // Register type
+            ASSERT_EQ(participant_->register_type(type_), ReturnCode_t::RETCODE_OK);
```

Comment:
These changes have not been applied to the Fast RTPS deprecated API. Consequently, the FastRTPS blackbox test is failing due to the failure creating the participant.